src/main/java/org/synyx/urlaubsverwaltung/publicholiday/PublicHolidaysServiceImpl.java

```diff
@@ -80,9 +87,32 @@ private DayLength getHolidayDayLength(WorkingTimeSettings workingTimeSettings, L
     }
 
     private Set<Holiday> getHolidays(final LocalDate from, final LocalDate to, FederalState federalState) {
-        return getHolidayManager(federalState)
-            .map(holidayManager -> holidayManager.getHolidays(from, to, federalState.getCodes()))
+
+        final Set<Holiday> holidays = getHolidayManager(federalState)
+            .map(holidayManager -> holidayManager.getHolidays(from, to, PUBLIC_HOLIDAY, federalState.getCodes()))
             .orElseGet(Set::of);
+
+        final DateRange requestRange = new DateRange(from, to);
+
+        final LocalDate christmasFrom = LocalDate.of(from.getYear(), Month.DECEMBER, 24);
+        if (requestRange.isOverlapping(new DateRange(christmasFrom, christmasFrom))) {
+            holidays.add(new Holiday(christmasFrom, "CHRISTMAS_EVE", PUBLIC_HOLIDAY));
+        }
+        final LocalDate christmasTo = LocalDate.of(to.getYear(), Month.DECEMBER, 24);
+        if (requestRange.isOverlapping(new DateRange(christmasTo, christmasTo))) {
+            holidays.add(new Holiday(christmasTo, "CHRISTMAS_EVE", PUBLIC_HOLIDAY));
+        }
```

Comment:
mehrere jahre in der auswahl zu haben wird noch nicht der fall sein? könnte, wenn man abgefahrene statistiken einführt? wollen wir das jetzt schon beachten oder schieben? ^^"

from 2024-01-01 to 2026-12-31, da müssten dann drei heiligabende in holidays sein